src/kontrol/prove.py

```diff
@@ -978,101 +987,104 @@ def _create_cse_accounts(
-                            ],
-                        ),
-                    ],
-                ),
-                intToken(0),
-            )
-            lowest_bit_not_set = eqInt(
-                intToken(0),
-                KApply(
-                    '_&Int_',
-                    [
-                        intToken(1),
-                        lookup,
-                    ],
-                ),
+            string_contents = KVariable(field_name + '_S_CONTENTS', sort=KSort('Bytes'))
+            string_length = KVariable(field_name + '_S_LENGTH', sort=KSort('Int'))
+            string_structure = KApply(
+                'asWord',
+                [
+                    KApply(
+                        '_+Bytes__BYTES-HOOKED_Bytes_Bytes_Bytes',
+                        [
+                            string_contents,
+                            KApply(
+                                'buf',
+                                [
+                                    intToken(1),
+                                    KApply('_*Int_', [intToken(2), string_length]),
+                                ],
+                            ),
+                        ],
+                    )
+                ],
             )
-
-            new_account_constraints.append(mlEqualsTrue(lowest_bit_not_set))
-            new_account_constraints.append(mlEqualsTrue(length_byte_lt32))
-            new_account_constraints.append(mlEqualsTrue(length_byte_positive))
+            string_contents_length = eqInt(KEVM.size_bytes(string_contents), intToken(31))
+            string_length_lb = leInt(intToken(0), string_length)
+            string_length_ub = ltInt(string_length, intToken(32))
+
+            storage_map = KApply('_Map_', [map_item(intToken(field.slot), string_structure), storage_map])
+            new_account_constraints.append(mlEqualsTrue(string_contents_length))
+            new_account_constraints.append(mlEqualsTrue(string_length_lb))
+            new_account_constraints.append(mlEqualsTrue(string_length_ub))
+        # Processing of addresses
+        if field.data_type == 'address':
+            if field.slot in singly_occupied_slots:
+                # The offset must equal zero
+                assert field.offset == 0
+                # Create appropriate symbolic variable
+                field_variable = KVariable(field_name + '_ID', sort=KSort('Int'))
+                # Update storage map accordingly: ( field.slot |-> contract_account_variable ) STORAGE_MAP
+                storage_map = KApply('_Map_', [map_item(intToken(field.slot), field_variable), storage_map])
+                address_range_lb = leInt(intToken(0), field_variable)
+                address_range_ub = ltInt(field_variable, intToken(1461501637330902918203684832716283019655932542976))
+                new_account_constraints.append(mlEqualsTrue(address_range_lb))
+                new_account_constraints.append(mlEqualsTrue(address_range_ub))
```

Comment:
Addresses are assumed to be initialised by the constructor and are placed in storage explicitly.